aerosandbox/library/power_solar.py

```diff
@@ -140,6 +264,87 @@ def solar_flux_on_horizontal(
             incidence_angle_function(latitude, day_of_year, time, scattering)
     )
 
+def solar_flux_new(
```

Comment:
As before, this seems like a pure generalization of `solar_flux_on_horizontal`, so perhaps we just replace that with this?